noise/README.md

```diff
@@ -211,26 +211,30 @@ messages. We leverage this construct to transmit:
 
 1. the libp2p identity key along with a signature, to authenticate each party to
    the other.
-2. arbitrary data private to the libp2p stack. This facility is not exposed to
-   userland. Examples of usage include streamlining muxer selection.
-
-These payloads MUST be inserted into the first message of the handshake pattern
-**that guarantees secrecy**. In practice, this means that the initiator must not
-send a payload in their first message. Instead, the initiator will send its
-payload in message 3 (closing message), whereas the responder will send theirs
-in message 2 (their only message). It should be stressed, that the second
-message of the handshake pattern has forward secrecy, however the sender has not
-authenticated the responder, so this payload might be sent to any party,
-including an active attacker.
+2. extensions used by the libp2p stack.
+
+The extensions are inserted into the first message of the handshake pattern
+**that guarantees secrecy**. Specifically, this means that the initiator MUST NOT
+send extensions in their first message. 
+The initiator sends its extensions in message 3 (closing message), and the 
+responder sends theirs in message 2 (their only message). It should be stressed,
+that while the second message of the handshake pattern has forward secrecy, 
+the sender has not authenticated the responder yet, so this payload might be
+sent to any party, including an active attacker.
 
 When decrypted, the payload contains a serialized [protobuf][protobuf]
 `NoiseHandshakePayload` message with the following schema:
 
 ``` protobuf
+message NoiseExtensions {
```

Comment:
For security reasons such as DOS, do we want to specify a size limit or item limit on the extensions?

Reply:
We should, good catch.

The amount of data we want to transmit might be on the larger side (all of our protocols), but then this is the second flight, so in practice we're limited by the host's congestion controller: If we try to send too much data, it won't fit into the congestion window and then we have to wait 1 RTT for acknowledgements to arrive, which would nullify the benefit of using an extension in the first place.
I'd assume that sending more than 1 kB would get us into problems here. We should probably point out this problem in the spec. The size limit for DoS protection can be higher than this, I guess 4 or 8kb would be fine?